app/src/main/java/com/asfoundation/wallet/ui/gamification/LevelsAdapter.kt

```diff
@@ -4,25 +4,22 @@ import android.content.Context
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
-import com.appcoins.wallet.gamification.LevelModel
-import com.appcoins.wallet.gamification.LevelModel.LevelType
 import com.asf.wallet.R
-import com.asfoundation.wallet.promotions.PromotionClick
 import com.asfoundation.wallet.util.CurrencyFormatUtils
 import io.reactivex.subjects.PublishSubject
-import java.math.BigDecimal
 
 class LevelsAdapter(private val context: Context,
-                    private val hiddenLevels: List<LevelModel>,
-                    shownLevels: List<LevelModel>,
-                    private val amountSpent: BigDecimal, private val currentLevel: Int,
-                    private val nextLevelAmount: BigDecimal?,
                     private val currencyFormatUtils: CurrencyFormatUtils,
                     private val mapper: GamificationMapper,
                     private val uiEventListener: PublishSubject<Pair<String, Boolean>>) :
     RecyclerView.Adapter<LevelsViewHolder>() {
 
-  private var activeLevelList: MutableList<LevelModel> = shownLevels.toMutableList()
+  /**
+   * these fields are meant to be properly initialized before onCreateViewHolder is called.
+   * @see setLevelsContent
+   */
+  private var hiddenLevels: List<LevelItem> = emptyList()
```

Comment:
I guess you could make it mutable and val so it is consistent with activeLevelList?